### DISCUSSION TITLE
How would you interpret these results?

### DISCUSSION THREAD

**Discussion by tonyg-swat** [GpuZelenograd/memtest_vulkan#15] [Card-specific memtest_vulkan error logs]
First, thank you for memtest_vulkan -- it has been very helpful!

I am getting output that looks pretty consistently like this:

```
Error found. Mode NEXT_RE_READ, total errors 0x1C216 out of 0x1C000000 (0.02452774%)
Errors address range: 0x120288..=0x6FFFE2CB  iteration:1
values range: 0xFFFBD4B4..=0x0001140B   FFFFFFFF-like count:0    bit-level stats table:
         0x0 0x1  0x2 0x3| 0x4 0x5  0x6 0x7| 0x8 0x9  0xA 0xB| 0xC 0xD  0xE 0xF
SinglIdx                 |                 | 75k      21k    |   1             
   0x1? 2413      15k    |                 |                 |                 
TogglCnt    114k  828    |                 |                 |                 
1sInValu                 |       9   24 116| 3691202 28125646|9382 14k  17k 18k
   0x1?  16k 12k 86234482|1961 709  218  70|  14             |                 

Error found. Mode INITIAL_READ, total errors 0x21F0E out of 0x1C000000 (0.02959413%)
Errors address range: 0x70005008..=0xDFFFF6CB  iteration:69
values range: 0xFFFB8DDB..=0x0000703B   FFFFFFFF-like count:0    bit-level stats table:
         0x0 0x1  0x2 0x3| 0x4 0x5  0x6 0x7| 0x8 0x9  0xA 0xB| 0xC 0xD  0xE 0xF
SinglIdx                 |                 |100k      22k    |   2             
   0x1? 2712      12k    |                 |                 |                 
TogglCnt    137k 1421    |                 |                 |                 
1sInValu                 |            3  12|  76 307  9992683|6059 10k  16k 20k
   0x1?  22k 20k  16k 10k|60912791 1010 308|  69  20    4    |                 
```

I've looked at the README, at the how-to-interpret-the-results section, and none of the scenarios seem quite to fit. How would you interpret the distributions shown here?

**Comment by galkinvv** [comment]
What is your GPU model? This sometimes helps better analyzing the problem (for example I have some strange failures on GTX970/980 GPUs)

It seems that most of your errros are single bit, some are 2-bit, and never more. This mostly exludes data-inversion line errors.
And the problematic bit indices are always similar - 0x08, 0x0A, 0x0C, 0x10, 0x12 (5 different bits, never the others). So "some datalines are often failing" while "other data lines are always fine". This mostly exludes address bus problems.

So, this looks like a variant of a "multi-bit transmission error", where 1-or-2 of 5 bits are randomly mistramsmitted. Looks like a controller inside a memory chip or GPU chip is introducing errors (or maybe can't handle clocks/timings).

Or **really extreme** dirt near balls connecting data lines of chips to PCB, so huge that it nearly equally touch 5 lines, but this is less realistic.

**Reply by tonyg-swat** [reply]
Thanks for the analysis. This is all on a stock Lenovo Thinkpad X1 Carbon Gen 10 (model 21CBCTO1WW, "LENOVO_MT_21CB_BU_Think_FM_ThinkPad X1 Carbon Gen 10"). The graphics card is an Intel Iris Xe, driver version 31.0.101.4502:

```
Device PCI\VEN_8086&DEV_46A8&SUBSYS_22E717AA&REV_0C\3&11583659&1&10 was configured.

Driver Name: oem154.inf
Class Guid: {4d36e968-e325-11ce-bfc1-08002be10318}
Driver Date: 12/03/2022
Driver Version: 31.0.101.3959
Driver Provider: Intel Corporation
Driver Section: iADLPD_w10_DS
Driver Rank: 0xCF0001
Matching Device Id: PCI\VEN_8086&DEV_46A8&SUBSYS_22E717AA
Outranked Drivers: oem53.inf:PCI\VEN_8086&DEV_46A8&SUBSYS_22E717AA:00CF0001 display.inf:PCI\CC_0300:00FB2008
Device Updated: true
Parent Device: ACPI\PNP0A08\0
```

**Reply by galkinvv** [reply]
Oh, I have to note that while the analysis above is quite generic, really 99% of my error research was targeting descrete GPUs. And their **G**DDR memory has very different error patterns compared to the DDR.

And personally I never seen an errors in memtest_vulkan for the integrated GPU that uses shared system memory, so chances are that I'm wrong with something regarding to them.

Since integrated GPU shares memory with overall system - I can suggest running standard memtest for system memory to see if it detect any problems.

**Reply by tonyg-swat** [reply]
Thanks, good idea re a normal memtest -- I will also do that.
